src/Sylius/Component/Product/Model/Product.php

```diff
@@ -189,7 +194,7 @@ public function setMetaDescription($metaDescription)
      */
     public function isAvailable()
     {
-        return new \DateTime() >= $this->availableOn;
+        return new \DateTime() >= $this->availableOn && (!$this->availableTo || $this->availableTo >= new \DateTime());
```

Comment:
you should invert your comparaison order and make it like: 
```
new \DateTime() <= $this->availableTo;
```

to keep the same logic with the check before